Build failure with CLANG 18 and Ubuntu 24.04

With Ubuntu 24.04, CLANG 18.1 produces
```
/home/sbellock/repo/steven-bellock/libspdm/os_stub/mbedtlslib/mbedtls/library/bignum.c:1395:29: error: variable 't' set but not used [-Werror,-Wunused-but-set-variable]
 1395 |     mbedtls_mpi_uint c = 0, t = 0;
 
 /home/sbellock/repo/steven-bellock/libspdm/library/spdm_responder_lib/libspdm_rsp_encap_challenge.c:178:5: error: variable '__debug_code_local' set but not used [-Werror,-Wunused-but-set-variable]
  178 |     LIBSPDM_DEBUG_CODE(
```

--- a/include/hal/library/debuglib.h
+++ b/include/hal/library/debuglib.h
@@ -106,10 +106,7 @@ extern void libspdm_debug_assert(const char *file_name, size_t line_number,
 #if LIBSPDM_DEBUG_BLOCK_ENABLE
 #define LIBSPDM_DEBUG_CODE(expression) \
     do { \
-        uint8_t __debug_code_local; \
         expression \
-            __debug_code_local = 0; \
-        __debug_code_local++; \
     } while (false)
 #else
 #define LIBSPDM_DEBUG_CODE(expression)

